cms/djangoapps/contentstore/views/tests/test_videos.py

```diff
@@ -228,6 +234,84 @@ def test_get_json(self):
                 convert_video_status(original_video)
             )
 
+    @ddt.data(True, False)
+    @patch('openedx.core.djangoapps.video_config.models.VideoTranscriptEnabledFlag.feature_enabled')
+    def test_get_json_transcripts(self, is_video_transcript_enabled, video_transcript_feature):
+        """
+        Test that transcripts are attached based on video transcript feature enablement.
+        """
+        video_transcript_feature.return_value = is_video_transcript_enabled
+        response = self.client.get_json(self.url)
+        self.assertEqual(response.status_code, 200)
+        response_videos = json.loads(response.content)['videos']
+        self.assertEqual(len(response_videos), len(self.previous_uploads))
+
+        expected_keys = ['edx_video_id', 'client_video_id', 'created', 'duration', 'status', 'course_video_image_url']
+        if is_video_transcript_enabled:
+            expected_keys += ['transcripts']
+
+        for response_video in response_videos:
+            self.assertEqual(set(response_video.keys()), set(expected_keys))
+
+    @ddt.data(
+        ([], {}),
+        (
+            [
+                {
+                    'video_id': 'test1',
+                    'language_code': 'en',
+                    'file_name': 'edx101.srt',
+                    'file_format': 'srt',
+                    'provider': 'Cielo24'
+                }
+            ],
+            {
+                'en': 'English'
+            }
+        ),
+        (
+            [
+                {
+                    'video_id': 'test1',
+                    'language_code': 'en',
+                    'file_name': 'edx101_en.srt',
+                    'file_format': 'srt',
+                    'provider': 'Cielo24'
+                },
+                {
+                    'video_id': 'test1',
+                    'language_code': 'es',
+                    'file_name': 'edx101_es.srt',
+                    'file_format': 'srt',
+                    'provider': 'Cielo24'
+                }
+            ],
+            {
+                'en': 'English',
+                'es': 'Spanish'
+            }
+        )
+    )
+    @ddt.unpack
+    @patch('openedx.core.djangoapps.video_config.models.VideoTranscriptEnabledFlag.feature_enabled')
+    def test_get_transcripts(self, transcripts_data, expected_transcripts, video_transcript_feature):
+        """
+        Test that transcripts are attached based on video transcript feature enablement.
+        """
+        is_video_transcript_enabled = True
+        video_transcript_feature.return_value = is_video_transcript_enabled
```

Comment:
I think we don't need `is_video_transcript_enabled` variable, we can directly set the return_value to True